sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/_connection.py

```diff
@@ -91,18 +90,41 @@ class Connection:  # pylint:disable=too-many-instance-attributes
      and 1 for transport type AmqpOverWebsocket.
     """
 
-    def __init__(self, endpoint, **kwargs):  # pylint:disable=too-many-statements
-        # type(str, Any) -> None
+    def __init__( # pylint:disable=too-many-locals,too-many-statements
+            self,
+            endpoint: str,
+             *,
+            container_id: Optional[str] = None,
+            max_frame_size: int = MAX_FRAME_SIZE_BYTES,
+            channel_max: int = MAX_CHANNELS,
+            idle_timeout: Optional[float] = None,
+            outgoing_locales: Optional[List[str]] = None,
+            incoming_locales: Optional[List[str]] = None,
+            offered_capabilities: Optional[List[str]] = None,
+            desired_capabilities: Optional[List[str]] = None,
+            properties: Optional[Dict[str, Any]] = None,
+            allow_pipelined_open: bool = True,
+            idle_timeout_empty_frame_send_ratio: float = 0.5,
+            idle_wait_time: float = 0.1,
+            network_trace: bool = False,
+            transport_type: TransportType = TransportType.Amqp,
+            http_proxy: Optional[Dict[str, Any]] = None,
+            socket_timeout: Optional[float] = None,
+            **kwargs: Any):  # pylint:disable=too-many-statements
         parsed_url = urlparse(endpoint)
+
+        if parsed_url.hostname is None:
```

Comment:
This is because urlparse can return None for the hostname, so mypy was complaining that hostname is None | str. At this point in the code we shouldn't have it be None. I dont think something similar is in uAMQP.

Reply:
update:- uAMQP also raises a `ValueError` https://github.com/Azure/azure-uamqp-python/blob/c85efcd12c249999eb8a1064b7d4fd8c7715c780/uamqp/address.py#L143